### PR TITLE
WIP implement mentions + findUser falls back to webfinger

### LINKED ISSUE #27: Implement mentions
Mentioning user should work in same way as hashtags works

1. detect mentions in text in same way as for hashtags https://github.com/autogestion/pubgate-philip/blob/master/philip/src/components/Publish.svelte#L11
mention format is `@username@domain` 

2. wrap it in html, same as mastodon does (so it will be displayed at mastodon servers correctly
html format is `<span '
                       'class="h-card"><a'
                       'href="<user_id>" '
                       'class="u-url mention">@<span>username</span></a></span></p>'`
to get `user_id`, it should be fetched from mentioned user server, by using FollowYourNose principle ( https://domain/@username) or by using webfinger (to be compatible with pleroma, as described in #26 )

3. Create mention sub-object, same as for hashtag
https://github.com/autogestion/pubgate-philip/blob/master/philip/src/components/Publish.svelte#L27
   sub-object format                 `{'href': '<user_id>',
                     'name': '@username@domain',
                     'type': 'Mention'}`

4. Add to Create object `cc` attribute, same as when making a comment
`'cc': [user_id]`

Final output, sent to outbox, should be
`{
    "type": "Create",
    "cc": [<user_id>]
    "object": {
        "type": "Note",
        "content": "<span '
                       'class="h-card"><a'
                       'href="<user_id>" '
                       'class="u-url mention">@<span>username</span></a></span></p>",
        "tag": [
			{
				"type": "Mention",
				"href": "<user_id>",
				"name": "@username@domain"
			},        
        ]
    }
}`

`

### PR DESCRIPTION
**Pull request by traumschule**
see #26, #27

1. [X] match @user@domain or @user (base_url is used)
1. [X] wrap in html
1. [X] create mention object
1. [X] add cc
1. [ ] test search with non-pgi

Note: the output specification requires `"cc": [<user_id>]`. When mentioning multiple users backend raised 500 because the cc list was inserted as an array. Hence line 46 in *Publish.svelte* is `ap_object.cc = mentions.map(m => m.href);` instead of `ap_object.cc = [mentions.map(m => m.href)];`

### CODE REVIEW

File: philip/src/components/Publish.svelte [hunk 1]
```diff
@@ -27,14 +39,17 @@
...
       .map(v => v.trim())
       .map(getHashTag);
+    let mentions = getAllMentions(content)
+      .map(m => m.trim())
+      .map(m => getMention(m, getUserId(m)));
 
-    const data = wrapHashTagsWithLink(wrapLinksWithTags(content));
-
-    let ap_object = getCreateObject(data, tags);
+    const data = wrapMentions(wrapHashTagsWithLink(wrapLinksWithTags(content)));
```
Comment (autogestion) [inline philip/src/components/Publish.svelte]: :)))

**Comment by autogestion** [comment]
Screenshot_39
tested mentions. looks like domain part should be cropped. Or maybe put the hole username (with domain) inside `<span>$2</span>`

**Comment by autogestion** [comment]
and renders incorrect url in cc (which will be used by server to notify person which was mentioned)
Screenshot_41

**Comment by traumschule** [comment]
thanks for testing! needs some more :)

**Comment by autogestion** [comment]
Have tested mentions, works fine
Screenshot_42
Mentioned user do not follow test acc, still got post where he is mentioned


Same time, fallback to webginger failed when tried to search for `putinhujlo@pleroma.site`

**Pull request merged**